src/cloud/project/magento-app-php-application.md

```diff
@@ -41,96 +41,19 @@ php -m
 
--  Default extensions:
-   -  `bcmath`
-   -  `bz2`
-   -  `calendar`
-   -  `exif`
-   -  `gd`
-   -  `gettext`
-   -  `intl`
-   -  `mysqli`
-   -  `pcntl`
-   -  `pdo_mysql`
-   -  `soap`
-   -  `sockets`
-   -  `sysvmsg`
-   -  `sysvsem`
-   -  `sysvshm`
-   -  `opcache`
-   -  `zip`
+-  **Required extensions** install automatically and cannot be removed.
+-  **Default extensions** install by automatically. You can disable them as needed.
+-  **Optional extensions**–You can add or disable these extensions as needed.
 
--  Extensions that are installed and cannot be uninstalled:
-   -  `ctype`
-   -  `curl`
-   -  `date`
-   -  `dom`
-   -  `fileinfo`
-   -  `filter`
-   -  `ftp`
-   -  `hash`
-   -  `iconv`
-   -  `json`
-   -  `mbstring`
-   -  `mysqlnd`
-   -  `openssl`
-   -  `pcre`
-   -  `pdo`
-   -  `pdo_sqlite`
-   -  `phar`
-   -  `posix`
-   -  `readline`
-   -  `session`
-   -  `sqlite3`
-   -  `tokenizer`
-   -  `xml`
-   -  `xmlreader`
-   -  `xmlwriter`
-
--  Extensions that can be installed and uninstalled as needed:
-   -  `bcmath`
-   -  `bz2`
-   -  `calendar`
-   -  `exif`
-   -  `gd`
-   -  `geoip`
-   -  `gettext`
-   -  `gmp`
-   -  `igbinary`
-   -  `imagick`
-   -  `imap`
-   -  `intl`
-   -  `ioncube`
-   -  `ldap`
-   -  `mailparse`
-   -  `mcrypt`
-   -  `msgpack`
-   -  `mysqli`
-   -  `oauth`
-   -  `opcache`
-   -  `pdo_mysql`
-   -  `propro`
-   -  `pspell`
-   -  `raphf`
-   -  `recode`
-   -  `redis`
-   -  `shmop`
-   -  `soap`
-   -  `sockets`
-   -  `sodium`
-   -  `ssh2`
-   -  `sysvmsg`
-   -  `sysvsem`
-   -  `sysvshm`
-   -  `tidy`
-   -  `xdebug`
-   -  `xmlrpc`
-   -  `xsl`
-   -  `yaml`
-   -  `zip`
-   -  `pcntl`
+| Default extensions | Required extensions | Optional extensions |
+|--------------------|---------------------|---------------------|
+| `ctype`<br> `curl`<br> `date`<br> `dom`<br> `fileinfo`<br> `filter`<br> `ftp`<br> `hash`<br> `iconv`<br> `json`<br> `mbstring`<br> `mysqlnd`<br>  `openssl`<br> `pcre`<br> `pdo`<br> `pdo_sqlite`<br> `phar`<br> `posix`<br> `readline`<br> `session`<br> `sqlite3`<br> `tokenizer`<br> `xml`<br> `xmlreader`<br> `xmlwriter`<br> | `ctype`<br> `curl`<br> `date`<br> `dom`<br> `fileinfo`<br> `filter`<br> `ftp`<br> `hash`<br> `iconv`<br> `json`<br> `mbstring`<br> `mysqlnd`<br> `openssl`<br> `pcre`<br> `pdo`<br> `pdo_sqlite`<br> `phar`<br> `posix`<br> `readline`<br> `session`<br> `sqlite3`<br> `tokenizer`<br> `xml`<br> `xmlreader`<br> `xmlwriter`<br> | `bcmath`<br> `bz2`<br> `calendar`<br> `exif`<br> `gd`<br> `geoip`<br> `gettext`<br> `gmp`<br> `igbinary`<br> `imagick`<br> `imap`<br> `intl`<br> `ioncube`<br> `ldap`<br> `mailparse`<br> `mcrypt`<br> `msgpack`<br> `mysqli`<br> `oauth`<br> `opcache`<br> `pdo_mysql`<br> `propro`<br> `pspell`<br> `raphf`<br> `recode`<br> `redis`<br> `shmop`<br> `soap`<br> `sockets`<br> `sodium`<br> `ssh2`<br> `sysvmsg`<br> `sysvsem`<br> `sysvshm`<br> `tidy`<br> `xdebug`<br> `xmlrpc`<br> `xsl`<br> `yaml`<br> `zip`<br> `pcntl`<br> |
```

Comment:
I think it is best to clarify here. I don't know why any extension from the Required column would be duplicated anywhere else because you note that they cannot be removed.

Reply:
Restored original column headings as required and optional didn't convey correct meaning.

Reply:
Restored original column headings as required and optional didn't convey correct meaning.